indexing-service/src/test/java/org/apache/druid/indexing/overlord/supervisor/SupervisorManagerTest.java

```diff
@@ -434,6 +438,69 @@ public void testCreateSuspendResumeAndStopSupervisor()
     Assert.assertTrue(manager.getSupervisorIds().isEmpty());
   }
 
+  @Test
+  public void testGetActiveSupervisorIdForDatasourceWithAppendLock()
```

Comment:
there is a lot of fine-grained mocking here that makes the test aware of each and every call being made in the implementation. We should refactor the code so we can avoid such mocking.